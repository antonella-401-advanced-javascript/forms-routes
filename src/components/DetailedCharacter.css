.DetailedCharacter {
  width: 80%;
  max-width: 500px;
  margin: 50px auto;
  background-color: #01092ba6;
  border-radius: 5px;

  div {
    width: 70%;
    margin: auto;
    justify-content: center;
  }

  img {
    width: 100%;
    max-width: 400px;
    max-height: 390px;
    border-radius: 3px;
    margin: 7px auto;
  }
  p {
    text-align: center;
    color: whitesmoke;
    font-weight: bold;
  }
}