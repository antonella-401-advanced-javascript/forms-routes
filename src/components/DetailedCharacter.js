import React from 'react';
import PropTypes from 'prop-types';
import styles from './DetailedCharacter.css';

const DetailedCharacter  = ({ character }) => {
  return (
    <div className={styles.DetailedCharacter}>
      <div>
        <img src={character.photoUrl}/>
      </div>
      <p>{character.name}</p>
      <p>Gender: {character.gender}</p>
      <p>Hair Color: {character.hair}</p>
      <p>Weapon/Attack: {character.weapon}</p>
      <p>Affiliation: {character.affiliation}</p>
    </div>
  );
};

DetailedCharacter.propTypes = {
  character: PropTypes.object
};

export default DetailedCharacter;