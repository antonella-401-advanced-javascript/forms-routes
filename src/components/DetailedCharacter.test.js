import React from 'react';
import { shallow } from 'enzyme';
import DetailedCharacter from './DetailedCharacter';

describe('DetailedCharacter component', () => {
  it('renders DetailedCharacter', () => {
    const testDetailedCharacter = 
      {
        photoUrl: 'https://www.outbrain.com/techblog/wp-content/uploads/2017/05/road-sign-361513_960_720.jpg',
        name: 'test2',
        gender: 'test',
        hair: 'test color',
        weapon: 'tests',
        affiliation: 'test?'
      };
    const wrapper = shallow(<DetailedCharacter character={testDetailedCharacter} />);
    expect(wrapper).toMatchSnapshot();
  });
});