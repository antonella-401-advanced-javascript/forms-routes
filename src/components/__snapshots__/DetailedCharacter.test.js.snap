// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailedCharacter component renders DetailedCharacter 1`] = `
<div
  className="DetailedCharacter"
>
  <div>
    <img
      src="https://www.outbrain.com/techblog/wp-content/uploads/2017/05/road-sign-361513_960_720.jpg"
    />
  </div>
  <p>
    test2
  </p>
  <p>
    Gender: 
    test
  </p>
  <p>
    Hair Color: 
    test color
  </p>
  <p>
    Weapon/Attack: 
    tests
  </p>
  <p>
    Affiliation: 
    test?
  </p>
</div>
`;
